frontend/src/sagas.ts

```diff
@@ -0,0 +1,42 @@
+import axios from 'axios';
+import { takeEvery, call, put } from 'redux-saga/effects';
+import { getTodos, postTodos } from './actions';
+import { ToDoItem } from './types';
+
+export function* watchGetTodos() {
+  yield takeEvery(getTodos.request, workGetTodos);
+}
+export function* watchPostTodos() {
+  yield takeEvery(postTodos.request, todos);
+}
+
+function* workGetTodos(): Generator {
+  try {
+    const response: any = yield call(getRequest);
+    yield put(getTodos.success(response.data));
+  } catch (err) {
+    yield put(getTodos.failure(err.data));
+  }
+}
+function* todos(action: ReturnType<typeof postTodos.request>): Generator {
+  try {
+    const response: any = yield call(postRequest, action.payload);
+    yield console.log(response);
+  } catch (err) {
+    yield put(postTodos.failure(err));
+  }
+}
+
+function getRequest() {
+  return axios({
```

Comment:
`axios.get<Type>(url).then(res => res.data)`

Reply:
No overload matches this call.

Reply:
const response: any = yield call(
  axios.get<any>('/api/todo').then((res) => res.data),
);